tests/utils/splitters.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from rdkit import Chem
+
+from skfp.utils.splitters import (
+    _create_scaffolds,
+    scaffold_train_test_split,
+    scaffold_train_valid_test_split,
+)
+
+
+@pytest.fixture
+def all_molecules() -> list[str]:
+    all_smiles: list[str] = [
+        "CCO",
+        "CCN",
+        "CCC",
+        "CCCl",
+        "CCBr",
+        "C1CC1",
+        "C1CCC1",
+        "C1CCCC1",
+        "C1CCCCC1",
+        "C1CC1.OCC",
+        "C1CCC1.OCC",
+        "CC.OCC",
+    ]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in all_smiles]
+
+
+@pytest.fixture
+def disconnected_molecules() -> list[str]:
+    disconnected_smiles: list[str] = ["C1CC1.OCC", "C1CCC1.OCC", "CC.OCC"]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in disconnected_smiles]
+
+
+@pytest.fixture
+def no_ring_molecules() -> list[str]:
+    no_ring_smiles: list[str] = ["CCO", "CCN", "CCC", "CCCl", "CCBr"]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in no_ring_smiles]
+
+
+@pytest.fixture
+def benzodiazepines() -> list[str]:
+    benzodiazepines_smiles: list[str] = [
+        "C1CN=C(C2=CC=CC=C2)N=C1",
+        "C1CN=C(C2=CC=CC=C2F)N=C1",
+        "C1CN=C(C2=CC=CC=C2Cl)N=C1",
+    ]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in benzodiazepines_smiles]
+
+
+@pytest.fixture
+def monosaccharides() -> list[str]:
+    monosaccharides_smiles: list[str] = [
+        "C(C1C(C(C(C(O1)O)O)O)O)O",
+        "C(C(C1C(C(C(C(O1)O)O)O)O)O)O",
+        "C(C(C(C(C(C=O)O)O)O)O)O",
+        "C(C(C(C(C(CO)O)O)O)O)O",
+        "C(C(C(C(C(CO)O)O)O)O)O",
+    ]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in monosaccharides_smiles]
+
+
+def test_train_test_split_total_molecule_count(all_molecules):
+    train_split, test_split = scaffold_train_test_split(
+        all_molecules, train_size=0.8, test_size=0.2
+    )
+    assert len(train_split) + len(test_split) == len(all_molecules)
```

Comment:
We can ensure precise lengths in all tests here, since we control the entire test data. Add `assert` for those.